lib/recog/matcher.rb

```diff
@@ -22,21 +27,26 @@ def match_banners(banners_file)
         reporter.increment_line_count
 
         line = line.to_s.unpack("C*").pack("C*").strip.gsub(/\\[rn]/, '')
-        extractions = nil
+        found_extractions = false
+
         fingerprints.each do |fp|
-          break if (extractions = fp.match(line))
+          extractions = fp.match(line)
+          if extractions
+            found_extractions = true
+            extractions['data'] = line
+            reporter.match "MATCH: #{extractions.inspect}"
```

Comment:
In testing this locally, I realized that the multi match functionality works but perhaps not in a way that allows us to deal with addressing the multiple matches issue in the fingerprints because `recog_match` does not make it obvious in the output whether or not there were multiple matches.  As written, it will output one line per matching fingerprint found against the current input string, but because `recog_match` can handle reading from STDIN or reading multiple banners from a file, the output very quickly becomes cluttered and you won't be able to tell what belongs to what or when multiple matches happen.  I'll shoot you a PR to correct this.